### PR TITLE
Material deposition

### PR DESCRIPTION
**Pull request by Rombur**
Here is finally the last PR to add material deposition in adamantine. I am not satisfied with the format of the input file and the current code is pretty slow but we can work on that later on.

### CODE REVIEW

**Comment by stvdwtt** [comment]
Great, I'll review this before our meeting on Tuesday.

File: application/adamantine.hh [hunk 1]
```diff
@@ -699,10 +717,45 @@ run(MPI_Comm const &communicator, boost::property_tree::ptree const &database,
...
+                << std::endl;
+    // Remove elements that have been activated
+    elements_to_activate.erase(elements_to_activate.begin() + activation_start,
+                               elements_to_activate.begin() + activation_end);
+    deposition_times.erase(deposition_times.begin() + activation_start,
+                           deposition_times.begin() + activation_end);
+    timers[adamantine::add_material].stop();
+
+    // Time can be different than time + time_step if an embedded scheme is
+    // used. Not that this is a problem when adding material because it means
```
[the remark this answers is not in the document]
Comment (Rombur) [reply application/adamantine.hh:757]: Yeah it should be a note. The problem is that with embedded scheme, the time step can be automatically reduced.

File: source/ThermalPhysics.templates.hh [hunk 2]
```diff
@@ -375,6 +377,96 @@ void ThermalPhysics<dim, fe_degree, MemorySpaceType,
...
+
+  // Communicate the results.
+  solution.compress(dealii::VectorOperation::min);
+
+  // Set the value to the newly create DoFs. Here we need to be careful with the
+  // hanging nodes. When there is a hanging node, the dofs at the vertices are
+  // "doubled": there is a dof associated to the coarse cell and a dof
+  // associated to the fine cell. The final value is decided by
+  // AffineConstraints. Thus, we need to make sure that the newly activated
+  // cells are at the same level than their neighbors.
```
[the remark this answers is not in the document]
Comment (Rombur) [reply source/ThermalPhysics.templates.hh:460]: No it's not

**Comment by Rombur** [comment]
This ready. I have a fix the MPI bug but it's based on this PR. I'll create the PR once this is merged.

**Review by stvdwtt — APPROVED** [review]

**Pull request merged**